internal/controller/userdeactivation_controller.go

```diff
@@ -0,0 +1,245 @@
+	Client     client.Client
+	Finalizers finalizer.Finalizers
+	Zitadel    *zitadel.Client
+}
+
+type userDeactivationFinalizer struct {
+	Client  client.Client
+	Zitadel *zitadel.Client
+}
+
+func (f *userDeactivationFinalizer) Finalize(ctx context.Context, obj client.Object) (finalizer.Result, error) {
+	log := logf.FromContext(ctx).WithName("userdeactivation-finalizer")
+
+	// Type assertion to get the user deactivation object.
+	userDeactivation, ok := obj.(*iammiloapiscomv1alpha1.UserDeactivation)
+	if !ok {
+		err := fmt.Errorf("unexpected object type %T, expected UserDeactivation", obj)
+		log.Error(err, "Type assertion failed")
+		return finalizer.Result{}, fmt.Errorf("type assertion failed: %w", err)
+	}
+	log.Info("Running finalizers", "userDeactivationName", obj.GetName(), "userDeactivationUID", obj.GetUID())
+
+	// Get the user reference from this UserDeactivation
+	userRef := userDeactivation.Spec.UserRef.Name
+
+	log.Info("Checking for other UserDeactivation objects", "userRef", userRef)
+	otherUserDeactivationsCount, err := f.countOtherUserDeactivations(ctx, userDeactivation, userRef)
+	if err != nil {
+		log.Error(err, "Failed to count other UserDeactivation objects")
+		return finalizer.Result{}, fmt.Errorf("failed to count other UserDeactivation objects: %w", err)
+	}
+
+	// If there are no other UserDeactivation objects for this user, reactivate the user
+	if otherUserDeactivationsCount == 0 {
+		log.Info("No other UserDeactivation objects found, reactivating user in Zitadel", "userRef", userRef)
+		err = f.Zitadel.ReactivateUser(ctx, userRef)
+		if err != nil {
+			log.Error(err, "Failed to reactivate user in Zitadel", "userRef", userRef)
+			return finalizer.Result{}, fmt.Errorf("failed to reactivate user in Zitadel: %w", err)
+		}
+		log.Info("Successfully reactivated user in Zitadel", "userRef", userRef)
+
+		// Update the User status to Active
+		user := &iammiloapiscomv1alpha1.User{}
+		err = f.Client.Get(ctx, client.ObjectKey{Name: userRef}, user)
+		if err != nil {
+			log.Error(err, "Failed to get User resource", "userRef", userRef)
+			return finalizer.Result{}, fmt.Errorf("failed to get User resource: %w", err)
+		}
+		user.Status.State = "Active"
+		err = f.Client.Status().Update(ctx, user)
+		if err != nil {
+			log.Error(err, "Failed to update User status", "userRef", userRef)
+			return finalizer.Result{}, fmt.Errorf("failed to update User status: %w", err)
+		}
+		log.Info("Successfully updated User status to Active", "userRef", userRef)
+	} else {
+		log.Info("Other UserDeactivation objects exist for user, skipping reactivation", "userRef", userRef)
+	}
+
+	return finalizer.Result{}, nil
+}
+
+// +kubebuilder:rbac:groups=iam.miloapis.com,resources=userdeactivations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=iam.miloapis.com,resources=userdeactivations/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=iam.miloapis.com,resources=userdeactivations/finalizers,verbs=update
```

Comment:
Can you confirm the RBAC this controller needs? By default kubebuilder will provide all permissions, but based on the reconcile function it looks like we're giving more permissions than necessary. 

Based on the reconcile function, it looks like it only needs `get/list/watch` on the `userdeactivations` resource, `get/update/patch` on the `userdeactivations/status` resource, and `update` on the `userdeactivations/finalizers` resource.